packages/react-reconciler/src/ReactChildFiber.js

```diff
@@ -120,16 +120,33 @@ function coerceRef(
         const componentName = getComponentName(returnFiber.type) || 'Component';
         if (!didWarnAboutStringRefs[componentName]) {
           if (warnAboutStringRefs) {
-            console.error(
-              'Component "%s" contains the string ref "%s". Support for string refs ' +
-                'will be removed in a future major release. We recommend using ' +
-                'useRef() or createRef() instead. ' +
-                'Learn more about using refs safely here: ' +
-                'https://fb.me/react-strict-mode-string-ref%s',
-              componentName,
-              mixedRef,
-              getStackByFiberInDevAndProd(returnFiber),
-            );
+            if (
+              element._owner &&
+              element._self &&
+              element._owner.stateNode !== element._self
+            ) {
+              console.error(
+                'Owner and self do not match for the string ref "%s". Support for ' +
```

Comment:
This doesn't really tell the user anything. They don't know what owner and self not matching means. How about:

_Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fixing this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://fb.me/react-strict-mode-string-ref%s_